spring-integration-core/src/main/java/org/springframework/integration/dsl/RouterSpec.java

```diff
@@ -197,31 +191,30 @@ void addMapping(Object key, NamedComponent channel) {
 		}
 
 		@Override
-		public void onApplicationEvent(ContextRefreshedEvent event) {
-			if (event.getApplicationContext() == getApplicationContext() && !this.initialized.getAndSet(true)) {
-				ConversionService conversionService = getConversionService();
-				if (conversionService == null) {
-					conversionService = DefaultConversionService.getSharedInstance();
+		protected void onInit() throws Exception {
```

Comment:
Just curious - you must have had a reason to defer this until the context refreshed.

If so, what has changed?

Reply:
As I said in the second part of the change - proper order for the comments registration.
And I guess previously that might not be a `LinkedHashMap` as well.

Now the `IntegrationFlow` bean for the sub-flow is processed before this `RouterMappingProvider`, therefore all the implicit channel beans got their bean names already before this `onInit()` is called.